modules/rostests/apitests/ntdll/NtStartProfile.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Test for NtStartProfile
+ * COPYRIGHT:   Copyright 2021 Thomas Faber (thomas.faber@reactos.org)
+ */
+
+#include "precomp.h"
+#include <versionhelpers.h>
+
+#define SIZEOF_MDL (5 * sizeof(PVOID) + 2 * sizeof(ULONG))
+typedef ULONG_PTR PFN_NUMBER;
+/* Maximum size that can be described by an MDL on 2003 and earlier */
+#define MAX_MDL_BUFFER_SIZE ((MAXUSHORT - SIZEOF_MDL) / sizeof(PFN_NUMBER) * PAGE_SIZE + PAGE_SIZE - 1)
+
+static BOOL IsWow64;
+static KAFFINITY SystemAffinityMask;
+static ULONG DummyBuffer[4096];
+
+/* The "Buffer[Offset]++;" should likely be within 128 bytes of the start
+ * of the function on any architecture we support
+ */
+#define LOOP_FUNCTION_SIZE_SHIFT 7
+#define LOOP_FUNCTION_SIZE (1UL << LOOP_FUNCTION_SIZE_SHIFT)
+C_ASSERT(LOOP_FUNCTION_SIZE == 128);
```

Comment:
Yes, unless I'm wrong about log2(128) being 7, since the goal here is to use 128 bytes, not "whatever 2**7 is".
I would love to `#define LOOP_FUNCTION_SIZE 128` and calculate `LOOP_FUNCTION_SIZE_SHIFT` based on that, but I'm aware of a simple way to do that.